### ISSUE TITLE
Reduce garbage collector's catalog load

### ISSUE THREAD

**Issue by alamb** [influxdata/influxdb_iox#5529]
# Usecase:
There is a garbage collector service in IOx ( see #5135  #5364 etc) that removes objects from the object store that doesn't have a corresponding entry in the postgres catalog.

However, the current implementation put a non trivial load on the catalog (e.g. 250-350 ops/second at times):

<img width="858" alt="Screen Shot 2022-08-31 at 6 04 40 PM" src="https://user-images.githubusercontent.com/490673/187915589-0e61bab8-85d7-42ff-9cf5-f18214457436.png">

Given that the catalog is a shared piece of infrastructure and its performance is critical to the rest of the system, reducing the garbage collector load is likely a good idea

# Current implementation
The current implementation is here: https://github.com/influxdata/influxdb_iox/blob/main/garbage_collector/src/checker.rs#L69-L72

Basically it does one giant object store `LIST` operation and then for each object calls the `get_by_object_store_id`  catalog function to see if that object is referenced in the catalog. 

# Posible Improvement Ideas

##  Batch up the catalog lookups
Idea is to look up some larger number (100?) of paths  per catalog transaction. Perhaps by changing the query to use an `IN` like:

```sql
select ... from parquet_file where id IN (id1, id2, id3, ...)
```

## Throttle the number of catalog operations per second
We could also add an explicit  throttle so it did no more than X catalog ops a second 

# Context

Internal slack reference: https://influxdata.slack.com/archives/CRK9M8L5Q/p1661983189198929

**Comment by aierui** [comment]
Hi @alamb , can you share the monitoring metrics(export json file) for grafana dashborad ?

**Comment by alamb** [comment]
>  Hi @alamb , can you share the monitoring metrics(export json file) for grafana dashborad ?

Hi @aierui  unfortunately I don't think I can do so easily as it has details of our internal infrastructure. However, I can share the query I am using:


```
sum by (op, result, pod)(rate(catalog_op_duration_seconds_count{job ="global-gc"}[1m]))
```

Where the pods running the garbage collector are tagged with a `job=global-gc-XX` attribute

**Comment by aierui** [comment]
Got it, thanks @alamb
